shared-workflows/tf-checkov-shared.yml

```diff
@@ -0,0 +1,27 @@
+## This github action runs Checkov against any terraform in the base directory. 
+## Checkov looks for common security related mistakes in terraform
+## Checkov generates findings and shows them in the actions tab.
+## More info about the action can be found here: https://www.checkov.io/4.Integrations/GitHub%20Actions.html
+## More info about what it checks can be found here: https://www.checkov.io/5.Policy%20Index/terraform.html
+
+name: Checkov
+on:
+  workflow_call:
+
+jobs:
+  checkov:
+    name: run checkov
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Test with Checkov
+        id: checkov
+        uses: bridgecrewio/checkov-action@master
```

Comment:
I would like to pin this to a version - @master terrifies me.

Reply:
they published VERY often and it often catches new stuff or refines the rules so that what it catches is more clear.   So I think I'd prefer to not have to PR this super often for updates.  What are you concerned with about @master ?

Reply:
Nothing prevents it from running arbitrary stuff, like delete the repo or exfil all my secrets or whatever.

Reply:
Nothing prevents a pinned version from doing that either unless one of us is reading through it every time.  I suppose there is some level of "eventually someone will catch on" but I kinda feel like we either trust it or we dont, they're doing some obfuscation in the backend with containers and stuff anyway.  Would you prefer I just remove checkov?  

I can also pin it to the latest release, but if we're concerned with external actions, we might be better off rolling our own and take on the full overhead of managing them.

Reply:
The long term intent is to roll out own, as mentioned in the other comment :)
I guess it's fine...
DO we have any way to monitor a feed for 'OH SNAP CHECKOV GOT HACKED' or something?